src/ContentGetter/PostsGenerator.php

```diff
@@ -12,62 +12,62 @@
  */
 class PostsGenerator implements ContentGetterContract
 {
-    /**
-     * Instance of the query.
-     *
-     * @var WP_Query
-     */
-    protected $query;
+	/**
+	 * Instance of the query.
+	 *
+	 * @var WP_Query
+	 */
+	protected $query;
 
-    /**
-     * Handles getting the content for the search query.
-     *
-     * @param WP_Query $query Instance of the query.
-     *
-     * @return array
-     */
-    public function getContent(WP_Query $query): array
-    {
-        $this->query = $query;
+	/**
+	 * Handles getting the content for the search query.
+	 *
+	 * @param WP_Query $query Instance of the query.
+	 *
+	 * @return array
+	 */
+	public function getContent(WP_Query $query): array
```

Comment:
I really do think, if I'm designing an implementation I want a `WP_REST_Request` here.

Reply:
Okay, then let's make that change.

Reply:
Ok, but you asked for upsides and downsides so I wrote the last paragraph of this comment: https://github.com/caldera-learn/rest-api-search/pull/6#issuecomment-392572533 and now I really feel like the change either:

* Strongly couples the interface to the REST API. 👎 
or
* We have to have another level of abstraction below it. Ok. I'll do that I guess.

Reply:
Change made
e2e1719